pkg/api/addons.go

```diff
@@ -540,10 +540,20 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		}
 	}
 
-	// Ensure cloud-node-manager is enabled on appropriate upgrades
-	if isUpgrade {
-		if i := getAddonsIndexByName(o.KubernetesConfig.Addons, CloudNodeManagerAddonName); i > -1 {
-			o.KubernetesConfig.Addons[i] = defaultCloudNodeManagerAddonsConfig
+	// Ensure cloud-node-manager and CSI components are enabled on appropriate upgrades
+	if isUpgrade && to.Bool(o.KubernetesConfig.UseCloudControllerManager) &&
+		common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.16.0") {
```

Comment:
Let's add a UT that validates that a 1.15-->1.16 upgrade w/ all three of these addons disabled initially is statically overwritten to enabled=true (which is what I interpret this back-compat logic to be doing)